### PR TITLE
refactor pkg_deb

### PR DESCRIPTION
**Pull request by aiuto**
Refactor debian building so it can be more easily owned by others.

blocked by #443

cc: @alexeagle

### CODE REVIEW

**Comment by aiuto** [comment]
This is ready again, now that the big reorganization PR is merged.

File: CODEOWNERS
Comment (nacl) [inline CODEOWNERS:8]: So, what's the exact policy on approvers here?  It seems that not having multiple CODEOWNERS could be risky if a change must be reviewed by a CODEOWNER.

File: distro/BUILD
Comment (nacl) [inline distro/BUILD:42]: What about inverting the structure a bit: `//pkg/deb/private`?  Would reduce the complexity of CODEOWNERS somewhat.

This is fine as-is though; just wondering your thoughts.

File: distro/testdata/BUILD.tpl
Comment (nacl) [inline distro/testdata/BUILD.tpl:1]: Was this missed in #443?

File: pkg/README.md
Comment (nacl) [inline pkg/README.md:44]: I wonder if this file could just be be called "deb.bzl" instead of "pkg_deb.bzl".

IMO, it's a bit redundant to have `pkg` said twice in the label.  As an example of prior art,`rpm.bzl` and `rpm_pfg.bzl` contain `pkg_rpm`.

**Comment by nacl** [comment]
We don't happen to have tests that exercise the distro tarball, would we?  A simple way to do this would be something that would build the distro, extract it somewhere, plant the tests inside the extracted location, and then run the tests.

**Review by nacl — CHANGES_REQUESTED** [review]
I think we need to confirm the policy on CODEOWNERs first so we prevent shooting ourselves in the foot.

LGTM otherwise, will approve once we come to a consensus on it and naming.

**Comment by aiuto** [comment]
> We don't happen to have tests that exercise the distro tarball, would we? A simple way to do this would be something that would build the distro, extract it somewhere, plant the tests inside the extracted location, and then run the tests.

Yes. The tests under distro do do that. They make sure we can build some things when using the tarball, and under a different repository name. 
We could probably add finer grained test in the future.

File: pkg/README.md
Comment (aiuto) [reply pkg/README.md:44]: Good point. I'll rename.

File: distro/testdata/BUILD.tpl
Comment (aiuto) [reply distro/testdata/BUILD.tpl:1]: Yes. But becaused of the backwrads compatibility, it still worked.
I tighented it up here.

File: distro/BUILD
Comment (aiuto) [reply distro/BUILD:42]: Too much in my head to think about that right now. That is time for another day.

File: CODEOWNERS
Comment (aiuto) [reply CODEOWNERS:8]: I think the policy works, in that you can't approve yourself.  But I don't actually know.
But.... once we do it, we can observe the behavior and correct it.

Comment (aiuto) [reply CODEOWNERS:8]: I find the only way to figure out some of the github interactions is to iteratively submit and see.

**Comment by aiuto** [comment]
> I think we need to confirm the policy on CODEOWNERs first so we prevent shooting ourselves in the foot.

What particular foot shooting are you thinking of?  
The current situation is that you and I are the only codeowners, and with the current branch protection policy, that means a PR from either of us is blocked on the other.

Adding extra owners will allow a domain expert to review and approve a change. If it is strictly to RPM or DEB packaging (or maybe even tar & zip) I welcome the ability to have someone else be on the hook to review things.
That's what this will do.

**Comment by nacl** [comment]
> What particular foot shooting are you thinking of? The current situation is that you and I are the only codeowners, and with the current branch protection policy, that means a PR from either of us is blocked on the other.
> 
> Adding extra owners will allow a domain expert to review and approve a change. If it is strictly to RPM or DEB packaging (or maybe even tar & zip) I welcome the ability to have someone else be on the hook to review things. That's what this will do.

I was mostly concerned regarding exclusivity to owners -- that if I were the only owner of the RPM code and I were unavailable, then a change couldn't be approved without editing the CODEOWNERs files.

Reading more closely, I see that this is more of an overlay and not really a problem.  Will approve momentarily.

File: distro/BUILD
Comment (nacl) [reply distro/BUILD:42]: Regardless of our current decision, given that this is an implementation detail, we can decide this later if we find it doesn't scale well.

File: distro/testdata/BUILD.tpl
Comment (nacl) [reply distro/testdata/BUILD.tpl:1]: Cool, thanks.

**Review by nacl — APPROVED** [review]
My concerns are addressed in comments and code updates.  LGTM.

**Pull request merged**
